pkg/prometheus/promcfg.go

```diff
@@ -2223,3 +2234,68 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	}
 	return cfg
 }
+
+func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *monitoringv1.PrometheusTracingConfig) (yaml.MapItem, error) {
+	cfg := yaml.MapSlice{}
+	objMeta := cg.prom.GetObjectMeta()
+
+	cfg = append(cfg, yaml.MapItem{
+		Key:   "client_type",
+		Value: tracingConfig.ClientType,
+	})
+
+	cfg = append(cfg, yaml.MapItem{
+		Key:   "endpoint",
+		Value: tracingConfig.Endpoint,
+	})
+
+	parsedFraction, err := strconv.ParseFloat(*tracingConfig.SamplingFraction, 64)
+	if err != nil {
+		return yaml.MapItem{}, err
+	}
+	if parsedFraction > 1 || parsedFraction < 0 {
```

Comment:
Prometheus is tolerant to out-of-bounds value (e.g. it will assume 0 for values < 0 and 1 for values > 1). We could the same here as a precaution.

Reply:
Should we just remove this validation and let prometheus take care of it @simonpasquier ?